### ISSUE TITLE
lint/nursery/useConsistentCurlyBraces warns when adding space between tags in JSX

### ISSUE THREAD

**Issue by pooya-badiee** [biomejs/biome#5365]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.12.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/11.0.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

Hi, Thank you for all your hard work on Biome 🙂
I was trying to add space between a link and a text in JSX. and I saw a warning for `lint/nursery/useConsistentCurlyBraces1, even though the  Curly Braces is important here the fix removes  it

```jsx
<View>
  <Text size="base" appearance="critical">
    It looks like we do not deliver to this region. See our shipping details
    <Link appearance="monochrome" to={`${shop.storefrontUrl}'pages/shipping-and-returns`} external>
      here
    </Link>{' '}
    or contact our Customer Care team for more information
  </Text>
</View>
```

![Image](https://github.com/user-attachments/assets/75b7e007-20b9-4281-929a-127b447ae0d3)

### Expected result

I expect biome to not show any warnings here and not remove the {" "} in the fix

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
Our playground is able to apply lint rule fixes, and is up to date with the `main` branch. It's the easiest way for us to figure out if the issue is already fixed.

**Comment by pooya-badiee** [comment]
@dyc3  I can not reproduce it in the playground. since the options is not there for `useConsistentCurlyBraces`, (Maybe I am not using the playground correctly?)
however, I pushed the code to a [repo](https://github.com/[pooya-badiee/biome-useConsistentCurlyBraces](https://github.com/pooya-badiee/biome-useConsistentCurlyBraces)) to reproduce it easier.
Also since it is not always clear what he issue is and why the {' '} matters. please check this react playground link:
https://playcode.io/2296416 

please let me know if there is any other context I can provide, Thank you.

**Comment by dyc3** [comment]
I was able to reproduce the behavior in 1.9.4, but this has already been fixed on `main`.

**Comment by fregante** [comment]
Duplicate of https://github.com/biomejs/biome/issues/4154
